migrate/sqlMigrator/sqlMigrator.go

```diff
@@ -0,0 +1,318 @@
+// Package sqlMigrator is used to migrate from Google datastore to the new SQL implementation
+package sqlMigrator
+
+import (
+	"context"
+	"fmt"
+	"github.com/dapperlabs/flow-playground-api/migrate/sqlMigrator/model"
+	"github.com/dapperlabs/flow-playground-api/migrate/sqlMigrator/storage/datastore"
+	sqlModel "github.com/dapperlabs/flow-playground-api/model"
+	"github.com/dapperlabs/flow-playground-api/storage"
+	"github.com/google/uuid"
+	"time"
+)
+
+// numErrors counts the errors that occur during migration
+var numErrors = 0
+
+func main() {
+	dstore := connectToDatastore()
+	sqlDB := connectToSQL()
+
+	projects := *getAllProjects(dstore)
+
+	fmt.Println("Starting migration for", len(projects), "projects...")
+	for _, proj := range projects {
+		if !proj.Persist {
+			continue
+		}
+		migrateProject(dstore, sqlDB, proj) // Includes transaction & script execution templates
+		migrateAccounts(dstore, sqlDB, proj.ID)
+		migrateUser(dstore, sqlDB, proj)
+		migrateScriptExecutions(dstore, sqlDB, proj.ID)
+		migrateTransactionExecutions(dstore, sqlDB, proj.ID)
+	}
+	fmt.Println("Migration finished with", numErrors, "errors")
+}
+
+func connectToDatastore() *datastore.Datastore {
+	// TODO: connect to actual datastore
+	store, err := datastore.NewDatastore(context.Background(), &datastore.Config{
+		DatastoreProjectID: "test-project", // "dl-flow",
+		DatastoreTimeout:   time.Second * 5,
+	})
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Connected to datastore")
+	return store
+}
+
+func connectToSQL() *storage.SQL {
+	// TODO: connect to the real postgres database
+	sqlDB := storage.NewPostgreSQL(&storage.DatabaseConfig{
+		User:     "newuser", // test db with newuser / password
+		Password: "password",
+		Name:     "postgres",
+		Port:     5432,
+	})
+	fmt.Println("Connected to SQL database")
+	return sqlDB
+}
+
+// getAllProjects returns a list of all projects in the datastore
+func getAllProjects(store *datastore.Datastore) *[]*model.InternalProject {
```

Comment:
You should paginate, this will become too much data that will need to be held in memory, possibly reaching the limit

Reply:
You can even create a wrapper function that knows how to handle pagination and then no matter which resource you are getting just make sure you are getting it in batch until the result is null which should indicate no more data.